api/src/glfs-fops.c

```diff
@@ -653,6 +732,56 @@ pub_glfs_stat(struct glfs *fs, const char *path, struct stat *stat)
     return ret;
 }
 
+GFAPI_SYMVER_PUBLIC_DEFAULT(glfs_fstatat, 11.0)
+int
+pub_glfs_fstatat(struct glfs_fd *pglfd, const char *path, struct stat *stat,
+                 int flags)
+{
+    int ret = -1;
+    xlator_t *subvol = NULL;
+    loc_t loc = {
+        0,
+    };
+    struct iatt iatt = {
+        0,
+    };
+    int reval = 0;
+    gf_boolean_t no_follow = 0;
+
+    DECLARE_OLD_THIS;
+    __GLFS_ENTRY_VALIDATE_FD(pglfd, invalid_fs);
+
+    GF_REF_GET(pglfd);
+
+    subvol = glfs_active_subvol(pglfd->fs);
+    if (!subvol) {
+        ret = -1;
+        errno = EIO;
+        goto out;
+    }
+retry:
+    no_follow = (flags & AT_SYMLINK_NOFOLLOW) == AT_SYMLINK_NOFOLLOW;
+    ret = glfs_resolve_at(pglfd->fs, subvol, pglfd->fd->inode, path, &loc,
```

Comment:
Looking at this method, one thing which pops up is, can we check if `glfs_resolve_at()` itself be used directly with `glfs_openat()` also, instead of `glfs_resolve_fd_loc()` ?

Reply:
Done